webapp/src/main/java/com/box/l10n/mojito/service/assetcontent/AssetContentService.java

```diff
@@ -63,23 +77,32 @@ public AssetContent createAssetContent(
     AssetContent assetContent = new AssetContent();
 
     assetContent.setAsset(asset);
-    assetContent.setContent(content);
+    assetContent.setContent(this.contentService.isPresent() ? "" : content);
     assetContent.setContentMd5(DigestUtils.md5Hex(content));
     assetContent.setBranch(branch);
     assetContent.setExtractedContent(extractedContent);
 
-    assetContent = assetContentRepository.save(assetContent);
+    assetContent = this.assetContentRepository.save(assetContent);
+    if (this.contentService.isPresent()) {
+      this.contentService.get().setContent(assetContent, content);
+    }
 
     return assetContent;
   }
 
   /**
-   * Proxy {@link AssetContentRepository#findOne()}.
+   * Proxy {@link AssetContentRepository#findById(Long)} ()}.
    *
-   * @param id {@link AssetContent#id}
+   * @param id {@link com.box.l10n.mojito.entity.AssetContent#getId()}
    * @return the asset content
    */
   public AssetContent findOne(Long id) {
-    return assetContentRepository.findById(id).orElse(null);
+    Optional<AssetContent> optionalAssetContent = this.assetContentRepository.findById(id);
+    if (optionalAssetContent.isPresent() && this.contentService.isPresent()) {
+      AssetContent assetContent = optionalAssetContent.get();
+      assetContent.setContent(this.contentService.get().getContent(assetContent).orElse(""));
```

Comment:
What will happen here if the content is stored in the database? The contentService will be null right, so the asset content will be returned as an empty string while it does exist in the database

Reply:
For l10n.asset-content-service.storage.type=s3, yes, that is correct.
If l10n.asset-content-service.storage.type=s3Fallback and the content does not exist in S3, then it will look at the database.

Reply:
Can you put a comment here explaining the process, making clear that if `S3` is selected with no fallback that an empty string will be used for the asset content if not found in S3

Reply:
Done